File: src/main/java/me/ikevoodoo/lifestealsmpplugin/commands/ReviveCommand.java
package me.ikevoodoo.lifestealsmpplugin.commands;

import me.ikevoodoo.lifestealsmpplugin.Configuration;
import me.ikevoodoo.lifestealsmpplugin.Utils;
import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.OfflinePlayer;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;

import java.util.List;
import java.util.UUID;

public class ReviveCommand implements CommandExecutor {

    @SuppressWarnings("deprecation")
    @Override
    public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
        if(sender.hasPermission("lssmp.revive")) {
            if(args.length > 0) {
                String person = args[0];
                if((person.equalsIgnoreCase("all")
                        || person.equalsIgnoreCase("@a")) && sender.hasPermission("lssmp.revive.all")) {
                    Configuration.getEliminations().forEach(Configuration::revive);
                    Bukkit.broadcastMessage(Utils.getFromText(ChatColor.GOLD + "Revived everyone!"));
                } else {
                    OfflinePlayer player = Bukkit.getOfflinePlayer(person);
                    List<UUID> eliminations = Configuration.getEliminations();
                    int index = eliminations.indexOf(player.getUniqueId());
                    if(index != -1) {
                        UUID id = eliminations.get(index);
                        Configuration.revive(id);
                        sender.sendMessage(ChatColor.GOLD + "Revived " + ChatColor.AQUA + player.getName());
                        return false;
                    }
                    sender.sendMessage(ChatColor.AQUA + player.getName() + ChatColor.RED + " is not eliminated.");
                }
            }
        }
        return false;
    }
}

File: src/main/kotlin/me/ikevoodoo/lssmp/commands/ReviveCommand.kt
package me.ikevoodoo.lssmp.commands

import me.ikevoodoo.lssmp.LSSMP
import me.ikevoodoo.lssmp.TranslationManager.Companion.getTranslation
import me.ikevoodoo.lssmp.utils.EliminationUtils.Companion.tryRevive
import me.ikevoodoo.lssmp.utils.ExtensionUtils.Companion.color
import me.ikevoodoo.lssmp.utils.ExtensionUtils.Companion.reset
import org.bukkit.Bukkit
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender

class ReviveCommand: CommandExecutor {
    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<out String>): Boolean {
        if(!sender.hasPermission("lssmp.revive")) {
            sender.getTranslation("commands.revive.no-perms")?.let { sender.sendMessage(it) }
            return true
        }

        if(args.isEmpty()) {
            sender.sendMessage((sender.getTranslation("commands.no-args") ?: "&cYou must specify at least player!").toString())
            return true
        }

        if(args.contains("all")) {
            LSSMP.ELIMINATION_STORAGE.clear()
            LSSMP.TARGET_STORAGE.clear()
            LSSMP.BAN_TIMES.clear()
            LSSMP.PLAYER_HEALTH_STORAGE.clear()
            LSSMP.TO_ELIMINATE.clear()

            LSSMP.INSTANCE.saveStorage()

            sender.sendMessage("&aAll players have been revived.".color())
            return true
        }

        args.forEach {
            val player = Bukkit.getOfflinePlayer(it)
            player.tryRevive()
            if(player.isOnline) player.player?.reset()
            else LSSMP.PLAYER_HEALTH_STORAGE[player.uniqueId.toString()] = "RESET"
        }

        sender.sendMessage("Revived ${args.size} players!")

        return true
    }
}